node-tests/automatic-new-tag.js

```diff
@@ -0,0 +1,103 @@
+const { expect } = require('chai');
+
+const { createBuilder, createTempDir } = require('broccoli-test-helper');
+
+const AutomaticNewTag = require('../lib/automatic-new-tag');
+
+let output;
+let input;
+
+async function buildFiles(files) {
+  input.write({
+    content: files,
+  });
+
+  const subject = new AutomaticNewTag(`${input.path()}/content`);
+  output = createBuilder(subject);
+
+  await output.build();
+
+  return output.read();
+}
+
+describe('Tag Generator', function() {
+  beforeEach(async () => {
+    input = await createTempDir();
+  });
+
+  afterEach(async () => {
+    try {
+      await input.dispose();
+    } finally {
+      // do nothing
+    }
+
+    if (output) {
+      await output.dispose();
+    }
+  });
+
+  it('should automatically create a new tag for the most recent 4 posts', async function() {
+    const files = await buildFiles({
+      'index.md': `---
+tags:
+  - a-tag-of-sorts
+authors:
+  - ghost
+date: Tue Jun 12 2018 17:50:59 GMT+0100 (IST)
+---
+# Hello world`,
+      'index2.md': `---
+tags:
+  - a-tag-of-sorts
+authors:
+  - ghost
+date: Tue Jun 11 2018 17:50:59 GMT+0100 (IST)
+---
+# Hello world`,
+      'index3.md': `---
+tags:
+  - a-tag-of-sorts
+authors:
+  - ghost
+date: Tue Jun 10 2018 17:50:59 GMT+0100 (IST)
+---
+# Hello world`,
+      'index4.md': `---
+tags:
+  - a-tag-of-sorts
+authors:
+  - ghost
+date: Tue Jun 09 2018 17:50:59 GMT+0100 (IST)
+---
+# Hello world`,
+      'index5.md': `---
+tags:
+  - a-tag-of-sorts
+authors:
+  - ghost
+date: Tue Jun 09 2018 17:50:59 GMT+0100 (IST)
+---
+# Hello world`
+});
+    expect(files['index.md']).to.equal(`---
+tags:
+  - a-tag-of-sorts
+  - new
+authors:
+  - ghost
+date: 'Tue Jun 12 2018 17:50:59 GMT+0100 (IST)'
+---
+# Hello world`)
+
+    expect(files['index5.md']).to.equal(`---
+tags:
+  - a-tag-of-sorts
+  - new
```

Comment:
this is the oldest one right? we don't expect this one to have the `new` tag right?

Reply:
Yeah totally. 🤔  That's strange, I thought that's what I did, but it can't have been because we didn't do the implementation for it.